src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -654,23 +655,30 @@ def _auto_select_batch_size(self) -> int:
         # train_micro_batch_size_per_gpu is used for throughput logging purposes
         # by default we try to use the batch size of the loader
         assert self.lightning_module is not None
-        batch_size = 1
```

Comment:
refactored so that I could use the shared `_get_dataloader_batch_size` function. Retrospectively, maybe not worth it. meh :(